### DISCUSSION TITLE
Temporary save layout hotkey

### DISCUSSION THREAD

**Discussion by buzhidaoist** [rxhanson/RectanglePro-Community#511] [Q&A] [answered]
Is there a way to set a keyboard shortcut hotkey or a pair of them that allows you to quickly "Save Current Layout" on all visible windows, defaulting to "bring windows to front" and "launch closed/minimized apps"*, such that using the hotkey immediately restores this layout?  

The use case is for people like me who frequently:
1. work with a certain window layout for a moderate period of time
2. the layout is temporary (i.e. high number of different layouts are used over time, with low-reuse, too many for assigning a persistent shortcut to each one)
3. at least two of the windows are of the same application, and with intermittent interruptions into other apps/windows (so it's a pain to manually restore the layout each time after an interruption)

I can imagine one of two solutions:
1. one shortcut to "quick-save" the current arrangement as a layout with the aforementioned defaults**, and another shortcut to activate it (e.g. ctrl+cmd+S to save, then ctrl+cmd+R to restore)
2. a pattern similar to vim macro recording, e.g. ctrl+cmd+S,G to save to layout register "G", then ctrl+shift+cmd+G to restore from "G"

**though if launch closed/minimized apps is enabled, the "quick-save" should ideally only save the currently visible and partially-visible windows to the layout definition

Please excuse me if this exists and I missed it, I tried searching but couldn't figure it out!

**Comment by rxhanson** [comment, marked as answer]
Thanks for the detailed feedback. At the moment, this flow doesn't exist, but I have had similar requests previously. Sounds like a copy-paste clipboard style of saving and applying layouts? I like this idea and will update the one I have on my backlog like this.

**Reply by buzhidaoist** [reply]
Yea clipboard style is good way to describe it! Or alternatively "saved checkpoint" style
